partiql-ast/src/main/java/org/partiql/ast/v1/Ast.java

```diff
@@ -0,0 +1,481 @@
+package org.partiql.ast.v1;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.partiql.ast.v1.expr.Expr;
+import org.partiql.ast.v1.expr.ExprAnd;
+import org.partiql.ast.v1.expr.ExprArray;
+import org.partiql.ast.v1.expr.ExprBag;
+import org.partiql.ast.v1.expr.ExprBetween;
+import org.partiql.ast.v1.expr.ExprCall;
+import org.partiql.ast.v1.expr.ExprCase;
+import org.partiql.ast.v1.expr.ExprCast;
+import org.partiql.ast.v1.expr.ExprCoalesce;
+import org.partiql.ast.v1.expr.ExprExtract;
+import org.partiql.ast.v1.expr.ExprInCollection;
+import org.partiql.ast.v1.expr.ExprIsType;
+import org.partiql.ast.v1.expr.ExprLike;
+import org.partiql.ast.v1.expr.ExprLit;
+import org.partiql.ast.v1.expr.ExprMatch;
+import org.partiql.ast.v1.expr.ExprNot;
+import org.partiql.ast.v1.expr.ExprNullIf;
+import org.partiql.ast.v1.expr.ExprOperator;
+import org.partiql.ast.v1.expr.ExprOr;
+import org.partiql.ast.v1.expr.ExprOverlay;
+import org.partiql.ast.v1.expr.ExprParameter;
+import org.partiql.ast.v1.expr.ExprPath;
+import org.partiql.ast.v1.expr.ExprPosition;
+import org.partiql.ast.v1.expr.ExprQuerySet;
+import org.partiql.ast.v1.expr.ExprSessionAttribute;
+import org.partiql.ast.v1.expr.ExprStruct;
+import org.partiql.ast.v1.expr.ExprSubstring;
+import org.partiql.ast.v1.expr.ExprTrim;
+import org.partiql.ast.v1.expr.ExprValues;
+import org.partiql.ast.v1.expr.ExprVarRef;
+import org.partiql.ast.v1.expr.ExprVariant;
+import org.partiql.ast.v1.expr.ExprWindow;
+import org.partiql.ast.v1.expr.PathStep;
+import org.partiql.ast.v1.expr.Scope;
+import org.partiql.ast.v1.expr.SessionAttribute;
+import org.partiql.ast.v1.expr.TrimSpec;
+import org.partiql.ast.v1.expr.WindowFunction;
+import org.partiql.ast.v1.graph.GraphDirection;
+import org.partiql.ast.v1.graph.GraphLabel;
+import org.partiql.ast.v1.graph.GraphMatch;
+import org.partiql.ast.v1.graph.GraphPart;
+import org.partiql.ast.v1.graph.GraphPattern;
+import org.partiql.ast.v1.graph.GraphQuantifier;
+import org.partiql.ast.v1.graph.GraphRestrictor;
+import org.partiql.ast.v1.graph.GraphSelector;
+import org.partiql.value.PartiQLValue;
+
+import java.util.List;
+import java.util.Map;
+
+// TODO docs for all factory methods
+public class Ast {
+    // Expr
+    @NotNull
+    public static ExprAnd exprAnd(Expr lhs, Expr rhs) {
+        return new ExprAnd(lhs, rhs);
+    }
```

Comment:
One potentially annoying thing I've noticed about these factory methods being written in Java is that they cannot be called from Kotlin with named parameters.

E.g. below results in an error "Named arguments are not allowed for non-Kotlin functions"
```
identifier(
    symbol = ctx.IDENTIFIER_QUOTED().getStringValue(),
    isDelimited = true
)
```

Wondering if we should actually have these factory methods written in Kotlin instead?

Reply:
I'll bring up next week w/ the team if these should be written in Kotlin.